---
title: "Use Microbiome Distribution Pattern and Abundancy Change to Predict and Treat Colonrectal Cancer "
author: "Maggie Chen"
date: "December 5, 2018"
output:
    github_document: default
bibliography: references.bib
csl: bioinformatics.csl
---

```{r setup, include = FALSE}
# Not show up this code and the outcome
# echo = FALSE hide all the codes for the graphs
knitr::opts_chunk$set(echo = FALSE)
```

# Introduction

Microbial communities have interacted with human communities continuously, and there are both benefits and harms can be done to human health by different bacteria. Not limited to the surface of the human body, internally bacteria also have an indispensable role in maintaining normal function and development of the human body [@sommer2016gut]. For instance, the microbies in the human gut represent a typical example of how humans can benefit from bacteria. The gut community supports metabolic activities, which help provide energy and nutrients for human's body. The gut community in the gut also help provide immune protection against alien microbes [@mcmurdie2013]. Within the adult gastrointestinal tract, the colon has the densest microbial community. In the luminal content there are than 10^11 bacteria per ml [@whitman1998prokaryotes]. However, the microbial community can also cause diseases. Examples include inflammatory bowel diseases, allergies, and colorectal cancer, which are due to changes in gut microbiome composition, and behavior disorder like depression [@de2012role]. 

Worldwide, colorectal cancer accountes for over a half million deaths annually, making it the third most commonly diagnosed malignancy [@parkin2005global]. There are two major stages of colorectal cancer development. Firstly, precancerous adenomatous polyps develop in the colon. These also known as the adenomas. In the second stage, the adenoma progresses into an invasive and metastatic cancerous tumor, called carcinoma [@levin2008screening]. The development of colorectal adenocarcinoma can be caused by the sequential or epigenetic mutation in the epithelial cells in the gastrointestinal tract. The mutation modifies the normal epithelial cells into hyperproliferative mucosa, which gives rise to benign adenoma [@pancione2012genetic]. The normal gastrointestinal epithelium differentiates into several epithelial colon lineages including pantech, goblet, and endocrine cells. They secrete mucosa to lubricate the colon and provide protection and anchor points for bacteria [@kosinski2007gene]. Thus, if a cancer-causing mutation changes the function of colon epithelial cells, there is a chance of altering the microbial community in the colon. Therefore, the abundance and composition of the gastrointestinal microbial community may serve as an indication of cancer development. Due to increase in early diagnosis, the deathes due to colorectal cancer have been decreasing in the United States. Current colon cancer screening and prediction methods include detecting gene mutation or methylation of specific genes that are related to colon cancer development [@kandimalla2017methylation]. Alternatively, to identify the prognostic micro RNA to predict the risk of developing colon cancer [@goossens2015microrna].    

Along with the rapid development of gene sequencing techniques, sequencing of microbial communities was also expanded extensively. High-throughput sequencing, also known as the Next Generation Sequencing (NGS) uses the non-Sanger-based methods to enable unprecedented DNA sequencing speed. This allows for novel biological sequencing applications. For example, 454, combines the emulsion PCR and pyrosequencing. For emulsion PCR, the single DNA amplification is a biases are reduced by avoiding the competition by multiple templates for a limited amount of DNA polymerase. Pyrosequencing, can be performed in parallel and produce a visual signal via computer system [@schuster2007next]. This next generation sequencing technique makes the DNA sequencing faster, which generates significantly more data for scientists to process. Thus, to process the massive amount of data, different methods have been developed. This can be applied along with the powerful statistical and graphical R programing language for various data analysis [@mcmurdie2013].

Thus, utilizing sequencing technologies and analysis tools, it is possible to examine colon cancer from a new perspective: correlation between microbiome composition and changes in cancer progression. For this paper, my primary question of interest is: what is the composition of the microbiomes that present in adenoma, carcinoma, and normal tissues. I will also examine whether there is a correlation between bacteria species and the personal variables like BMI, geographical locations and medication. I found that, for adenoma, carcinoma, and normal tissues, there is a similar microbial community composition, but differences in the abundance of each bacteria species. I also found that, bacteria can function in both directions, aid in cancer progression or help slow down the cancer progression. In addition, geographical regions and medication will also impact the microbial community in individuals, which can function as a possible prediction and guidance for colorectal cancer treatments. The knowledge of investigating the relationship between cancer and microbial communities in human bodies can be applied to other types of cancer, both in treatment, prediction and prevention. 

# Methods

## Study design

Zackular *et al*. examined 90 patients feces samples with 30 healthy, 30 colon carcinoma, and 30 colon adenoma to examine how the composition of the microbiome will be related to the colon cancer progression and whether the microbiome composition will help predict the adenoma and carcinoma lesions. Zackular et al. also include the factors including BMI, weight, height, geographical location of the patients to further investigate their correlation with microbial communities. For this paper, I used the sequence data and the metadata from Zackular et al.'s paper. Then analyzed the sequence data using DADA2, phlyoseq and R studio.

## Sample origin and sequencing

Approved by the University of Michigan Institutional Review Board and all subjects provided informed consent. Zackular et al. took samples from patients, that were 18 years of age or older, and who could tolerate 58ml of blood removal at two time points. There are a few more requirements the participants need to fulfill, including willing to complete a gFOBT kit, able to provide informed consent, also who had a colonoscopy and histologically confirmed colonic disease status. Zackular et al., excluded patients who is known to have HIV, chronic viral hepatitis, HNPCC, FAP, inflammatory bowel disease, as well as, any surgery radiation and chemotherapy for their current colorectal cancer or colonic adenoma. Fecal samples and the colonoscopies were collected and performed from four subjects of locations including, Toronto Canada; Boston Massachusetts, USA; Houston Texas, USA; and Ann Arbor Michigan, USA. The colonoscopies were followed by an endoscopic examination and patients who have no colonic abnormalities were designated as healthy. For examinations that revealed the presence of lesions resulted in a biopsy and subsequent diagnosis of adenoma or carcinoma. 

All the samples were collected as a whole evacuated stool a hat with no preservatives following the usual dietary and medication restriction for 24 hours. The gFOBT 6-Panel kit (Sensa HemocultII, Beckman-coulter) was prepared by patients immediately after the previous collection, from different areas of the stool. The stool was packaged with ice in an insulated box and shipped to the processing center to align with the gFOBT cards. Once the sample received, the feces samples we stored at -80ÂºC. The gFOBT was processed and interpreted as soon as the arrival. If 6 wells we positive on the gFOOBT kit, then the patients will be recorded as FOBT positive. At the time of stool collection, all the participants had intact colonic lesion., and samples were provided 1-2 weeks after the colons copy preparation. Overall, there were 90 stool samples received in total, with 30 healthy, 30 colon adenoma and 30 colon carcinoma samples [@zackular2014human]. The DNA extraction of samples used the Power-Soil-htp 96 well Soil DNA Isolation Kit (MO BIO Laboratories), and used EPMotion 5075 piptting system. Then the V4 regions of the 16S rRNA genes in each sample was amplified and sequenced using Illumina MiSeq Personal Sequencing platform [@zackular2014human]. 

## Computational

I used DADA to analyze and clean the data, then phyloseq and ggplot to visualize the results. Firstly, used a bash script to download the data set. Then I ran a quality check on a few sample data, which generated a quality profile on each of the samples. Then according to the sequence quality, I trimmed the data if they have more than three expected error, and has a quality lower than two. As a quality control, I created a table to compare the read counts before and after the trim to ensure I did not trim off too many sequences. After the trim, I build an error model for each sample to assess the probability of an unmatched base is an error or not, and a plot to check whether the error model matches the data. After it, I also deleted the duplicated sequences. Then as an essential step, I DADA to process the sequence data. DADA is an analysis include the error model and evaluates whether the unique sequence found is true or false. DADA eliminates the false positive results to provide the real variants and limit spurious sequence, to provide a high-resolution amplicon data [@callahan2016].

Then I listed the output of the DADA analysis on a table according to the variety of length. I also created a histogram to indicate the distribution of the sequence that is trimmed and denoised, to ensure that all the sequences are in acceptable length and quality. Using this output, I also removed all the chimeras, which are the cDNA sequence generated from two transcripts that is caused by polymerase stuck to another similar sequence. Using the sequence data, I assigned the taxonomy according to the supplied training set. Using the dada2 output, I charted a data set, which contains the metadata for each sample, Phylogeny, and taxonomy for each sequence. The phyloseq packges was used to import the information, and help analyze the phylogenetic sequence data clustered into Operational Taxonomic Units (OTU) [@mcmurdie2013]. I also melted the phyloseq data files to enable the ggplot function.


# Results
Firstly I was interested in the distribution of Phylum in different cancer progression stages, which includes normal, adenoma, and carcinoma. I found that in different types of fecal samples, they had a very similar composition of microbial communities, but there are differences in the abundance of different Phyla. The distribution pattern in three kinds of tissue samples are very different. The Bactericides abundance decreased, the normal samples has the highest abundance, then carcinoma and then adenoma samples. Euryarchaeota is in the opposite trending than the Bacteroides, the abundance increase in the order of normal, carcinoma, and adenoma. Fusobacteria is significantly more abundant in carcinoma samples. Proteobacteria is more abundant in adenoma and carcinoma samples than the normal samples (Figure 1). Next, I selected phyla that showed an interesting pattern across different sample types to further examine the Class distributions in non-normal samples and the correlation between Class and FOBT results. As a results, Bacterioidia has some abundance in both FOBT positive and negative samples from adenoma and carcinoma. In both adenoma and carcinoma samples, the FOBT sample had more abundant Fusobacteria than FOBT negative samples. The Clostridia and Negativicutes had an almost identical abundance in both FOBT positive and negative samples across the disease state. For Gammaproteobacteria, FOBT positive sample in both adenoma and carcinoma had a lower abundance (Figure 2). 

Then I selected out Gammaproteobacteria, Clostridia, Fusobacteriia, and Negativicutes to investigate their distribution in FBOT positive samples. I also investigate how gender will impact the abundance. For Fusobacteriia, there were mostly male samples only. Fusobacteria also had high abundance in carcinoma tissues, and lower in adenoma samples. For Gammaproteobacteria, in both adenoma and carcinoma samples, they had relatively similar abundance in both male and female samples. Clostridia had lower abundance in both male adenoma and carcinoma samples than female. This similar to the pattern of Negativicutes, in adenomas samples. However, Negativicutes had significantly higher abundance in male carcinoma than female (Figure 3). Giving Clostridia had a pattern consistency in adenoma and carcinoma sample, I then further investigated three specific Genus under Class Clostridia, Family Lachnospiraceae, which are Genus *Lactonifactor*, *Ruminococcus2*, *Anaerosporobacter*, with their abundance under the context of FOBT results and gender. In both male and female, FOBT results created a different abundance of the microbiome in adenoma and carcinoma samples. Within the Family Lachnospiraceae, I filtered several Genus that has relatively high abundance. *Anaerosporobacter* had a similar abundance for male and female in each of FOBT negative sample types, adenoma, carcinoma and normal. However, it is only found in female FOBT positive adenoma samples and male FOBT positive carcinoma samples. *Ruminococcus* was found in similar abundance in FOBT negative samples between male and females. However, in the FOBT positive samples, *Ruminococcus* was more abundant in both female adenoma and carcinoma samples than male (Figure 4). 

Since *Ruminococcus* had shown a different pattern between the FOBT positive and negative samples, and gender. I then further examined the geographical location's impact on *Ruminococcus*. For the adenoma samples with *Ruminococcus* detected, the patients are from Toronto or Ann Arbor, But mostly Toronto, ON, Canada. The carcinoma samples were mostly from Boston and Houston patients. Normal healthy samples were found in all four locations (Figure 5). Lastly, I was interested to see how different combinations of medication that contains the liptor will influence the *Ruminococcus* abundance in the samples. As the results, for the effect of medication in FOBT negative samples. Female patients who took Lipitor were found to have a relatively high abundance of *Ruminococcus*, and the average abundance is similar to each other. For male individuals, different combinations of treatment that contains Lipitor create a variation of *Ruminococcus* abundance. Individuals had only treated with Lipitor had the highest *Ruminococcus* mean abundance, then Lipitor with ezetrol and Lipitor with metformin and allopurinol had a decrease the *Ruminococcus* abundance. The treatment of Lipitor and novasc had significantly lowered the *Ruminococcus* abundance in male adenoma samples (Figure 6). 

 


```{r load-RData-knit}
# Use the load() function to produce load the 
# data sets and tables processes in the R script
# so don't have to run thorugh all the trim and DADA
# but still need to load necessary libaries for ggplot
# this load data set include the phyloseq, metadata, and melted data
load("Dataset.RData")
```
                    


```{r load-packages-and-data, include = FALSE}
# load general-use packages
library("dplyr")
library("tidyr")
library("knitr")
library("ggplot2")
library("tibble")

# this package allows for the easy inclusion of literature citations in our Rmd
# more info here: https://github.com/crsh/citr
# and here:
# http://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html
library("citr")

# These are the primary packages well use to clean and analyze the data
# this package needs to be installed from bioconductor -- it's not on CRAN
# see info here: https://benjjneb.github.io/dada2/dada-installation.html
library("dada2")

# This to export a fasta of our final denoised sequence variants
library("seqinr")

# To install this you have to install from GitHub
# See more info here: https://github.com/leffj/mctoolsr
# run this -- install.packages("devtools")
# and then this -- devtools::install_github("leffj/mctoolsr")
library("mctoolsr")

# And this to visualize our results
# it also needs to be installed from bioconductor
library("phyloseq")

```


```{r melt-tables}
# in case of the load data didn't include the metled data
# combine the metadata and the phyloseq to form the melted dataset
# which is used for ggplot in this report
melted_obj <- psmelt(phyloseq_obj)
```



```{r Phylum-disease-corrolation-barplot}
# To examine the difference of microbia composition
# under the context of Phylum, in different sample
# types and FOBT results
melted_obj %>%
  filter(Phylum != "NA") %>%
  ggplot(aes(x = Phylum,
             y = Abundance,
             fill = disease_stat,
             color = disease_stat)) +
  geom_col(position = "dodge") +
  ggtitle("Phylum and Disease State Corrolation") +
  labs(fill = "Disease State",
       color = "Disease State") + # re-label the coloring 
  xlab("Phlylum") +
  ylab("Abundance") +
  theme(axis.text.x = element_text(angle = 90,
                                   hjust = 1))
```

Figure 1. In different types of fecal samples, they have very similar composition of microbial communities, but there are differences in the abundance of different phylum. The distribution pattern in three kinds of tissue samples are very different. The Bacteroidetes abundance decreased in the order normal samples has the highest abundance, then carcinoma and then adenoma samples. Euryarchaeota is in the opposit trending than the Bacteroidets, the abundance increase in the order of normal, carcinoma, and adenoma. Fusobacteria is significantly more abundant in carcinoma samples. Proteibacteria is more abundant in adenoma and carcinoma samples than the normal samples. 


```{r top-Phylum-into-class-and-FOBT-boxplot}
# Take the most commonly found Phylums from previous figure
# Proteobacteria, Fusobacteria, Bacteroidetes, Firmicutes
# examine the class under them, and in different sample types
# and FOBT results
# the boxplot can help identify the outliers
# which gives a more accurate measurement

melted_obj %>%
  filter(Phylum %in% c("Proteobacteria", "Fusobacteria",
                       "Bacteroidetes", "Firmicutes")) %>%
  filter(Class != "NA") %>% #To exclude the NA samples
  filter(disease_stat != "normal") %>%
  ggplot(aes(x = Class,
             y = Abundance,
             color = FOBT.result)) +
  geom_boxplot(position = "dodge", width = 1) +
  ggtitle("FOTB Differentiate the Class Abundance in Variant Tissue Types ") +
  xlab("Class") +
  ylab("Abundance") +
  facet_grid(disease_stat ~ Phylum, scales = "free_x") +
  scale_y_log10() +
  theme(axis.text.x = element_text(angle = 40,
                                     hjust = 1))
```

Figure 2. Picked out four Phylum that has a different abundance in normal, adenoma, and carcinoma samples. Then examined their correlation between Class and FOBT results. There are several Class has significant high abundance in the FOBT positive samples. The box plot help to identify the outlines and the mean value of the abundance. As a results, Bacterioidia has the some abundance in both FOBT positive and negative samples from adenoma, and carcinoma. In both adenoma and carcinoma samples, the FOBT sample had more abundant Fusobacteriia than FOBT negative samples. The Clostridia and Negativicutes had a almost identical abundance in both FOBT positive and negative samples across the disease state. For Gammaproteobacteria, FOBT positive sample in both adenoma and carcinoma had lower abundance.

```{r class-tissue-sex-boxplot}
# Take the Classes that are highly found in adenoma and carcinoma tissue
# Gammaproteobacteria, Clostridia, Fusobacteriia, Negativicutes
# and the samples has the FOBT result positive
# Examine how aundance different in different genders
melted_obj %>%
  filter(Class %in% c("Gammaproteobacteria", "Clostridia",
                      "Fusobacteriia", "Negativicutes")) %>%
  filter(FOBT.result == "positive") %>%
  ggplot(aes(x = disease_stat,
             y = Abundance,
             color = sex)) +
  geom_boxplot(position = "dodge") +
  ggtitle("Top Four Class in FOBT Positive Tissue by Gender") +
  xlab("Disease State") +
  ylab("Abundance") +
  scale_y_log10() +
  facet_grid(~ Class) +
  theme(axis.text.x = element_text(angle = 40,
                                     hjust = 1))
```

Figure 3. Filtered out the Class that has a significant difference between samples from different cancer progression stages and FOBT results. Thus, to further analyze the correlation between microbiol community and the gender. I filtered out the samples that are FOBT positive and categorized into adenoma and carcinoma samples, colored the gender of the patients. For Fusobacteriia, there were mostly male samples only. Fusobacteria also had high abundance in carcinoma tissues, and lower in adenoma samples. For Gammaproteobacteria, in both adenoma and carcinoma samples, they had relatively similar abundance in both male and female samples. Clostridia had lower abundance in both male adenoma and carcinoma so maples than female. This similar to the pattern of Negativicutes, in adenomas samples. However, Negativicutes had significantly higher abundance in male carcinoma than female. 



```{r Clostridia-FOBT-sex-Genus-boxplot}
# For tissues that has blood clot in (FOBT positive),
# what are the Family found under Class Clostridia
# Family Lachnospiraceae, and Genus *Lactonifactor*, *Ruminococcus2*, *Anaerosporobacter*
# and their relationships with gender and cancer types


melted_obj %>%
  filter(Class %in% c("Clostridia")) %>%
  filter(Family %in% c("Lachnospiraceae")) %>%
  filter(Genus %in% c("Lactonifactor", "Ruminococcus2",
                      "Anaerosporobacter")) %>%
  ggplot(aes(x = Genus,
             y = Abundance,
             color = sex)) +
  geom_boxplot(position = "dodge") +
  facet_grid(disease_stat ~ FOBT.result, scales = "free_x") +
  ggtitle("Genus Variation in Disease and Gender") +
  xlab("Genus") +
  ylab("Abundance") +
  scale_y_log10() +
  theme(axis.text.x = element_text(angle = 40,
                                     hjust = 1))
```

Figure 4. In both male and female, FOBT results created a different abundance of mirobiome in adenoma and carcinoma samples. Within the Family Lachnospiraceae, I filtered serial Genus that has relatively high abundance. *Anaerosporobacter* had similar abundance for male and female in each of FOBT negative sample types, adenoma, carcinoma and normal. However, it is only found in female FOBT positive adenoma samples and male FOBT positive carcinoma samples. *Ruminococcus* was found in similar abundance in FOBT negative samples between male and females. But in the FOBT positive samples, *Ruminococcus* was more abundant in both female adenoma and carcinoma samples than male. 




```{r Ruminococcus-FOBTneg-loc-barplot}
# *Ruminococcus* has a interesting abundance in distribution in 
# FOBT positive tissue, but consistant in the FOBT negative samples
# between male and females, which might also be regarless of cancer stages
# From the FOBT negative tissues, how *Ruminococcus* is 
# distributed along with different disease state and locations

melted_obj %>%
  filter(Genus %in% c("Ruminococcus")) %>%
  filter(FOBT.result == "negative") %>%
  group_by(geo_loc_name, disease_stat) %>%
  summarize(sum_abund = sum(Abundance)) %>%
  filter(sum_abund > 0) %>%
  ggplot(aes(x = disease_stat,
             y = sum_abund,
             color = geo_loc_name,
             fill = geo_loc_name)) +
  geom_col(position = position_dodge()) +
  ggtitle("Genus Ruminococcus Distribution in Geographical Locations") +
  labs(fill = "Geographical Locations",
       color = "Geographical Locations") +
  xlab("Disease State") +
  ylab("Sum of Read Abundance")
```

Figure 5. For Genus *Ruminococcus*'s distribution in the FOBT negative samples, colored by the geographical locations. For the adenoma samples had *Ruminococcus* detected, the patients are from Toronto or Ann Arbor, But mostly Toronto, ON, Canada. The carcinoma samples were mostly from Boston and Houston patients. For normal healthy samples are more diverse and came from all four different locations.


```{r Ruminococcus-FOBTneg-non-normal-Medication}
# Under the Genus *Ruminococcus*, in the FOBT negative and non-normal tissue
# examine how the medication will impact the abundancy of *Ruminococcus*

melted_obj %>%
  filter(Genus %in% c("Ruminococcus")) %>%
  filter(FOBT.result == "negative") %>%
  filter(disease_stat != "normal") %>%
  filter(Medications != "unknown") %>%
  filter(grepl(Medications, pattern = "lipitor")) %>%
  ggplot(aes(x = Medications,
             y = Abundance,
             color = disease_stat)) +
  geom_boxplot(width = 0.3, position = position_dodge()) +
  ggtitle(" Genus Ruminococcus Distribution Under Different Treatments") +
  xlab("Medication Type") +
  ylab("Abundance") +
  labs(color = "Disease State") +
  scale_y_log10() +
  facet_wrap(disease_stat ~ sex, scales = "free_x") +
  theme(axis.text.x = element_text(angle = 90,
                                     hjust = 0.1))
```

Figure 6. To further investigate how *Ruminococcus* is impacted by the medication. The medication effects on *Ruminococcus* was only available in adenoma samples. This is focused on the effect of medication in FOBT negative samples. In female patients who had lipitor ins the treatment, they had  high abundance of *Ruminococcus*, and the average abundance is similar to each other. For male individuals, different combinations of treatment that contains lipitor creates a variation of *Ruminococcus* abundance. Individual had only treated with lipitor had the highest *Ruminococcus* mean abundance, then lipitor with ezetrol and lipitor with metformin and allupurinol had a decrease the *Ruminococcus* abundance. The treatment of lipitor and novasc had significantly lowered the *Ruminococcus* abundance in male adenoma samples. 

# Discussion

The result suggested that microbial communities have a vast variation of abundance depending on different disease state, gender, FOBT results, geographical locations, and medications. The variation of the microbial abundance, while have the sample phyla composition of the microbiome can sever as a method to distinguish and even predict the progression stages of the colon cancer. The Fusobacteria has a significant increase in abundance in the carcinoma samples, while in normal and adenoma tissue Fusobacteria remained at a very low concentration (Figure 1.). Thus, if patients have an increase in the Fusobacteria abundance, there might be a risk of developing colon cancer. Bacteroidetes is commonly found in the colon, human skin even healthy placenta [@mor2015trophoblast], and it is highly abundant in normal tissues and experience a decrease in abundance in adenoma and carcinoma samples (Figure 1). Thus, if the feces screening has found a decrease in the Bacterioidetes, then more test might be needed to further evaluate the risk of developing colon cancer. However, the phylum might be a categorize with higher uncertainties of which specific bacteria we are looking for and whether other factors plays a critical role in colon cancer. Thus, looking into the class underneath and whether Blood in the fecal samples will aid into the change of microbiome. The FOBT, fecal occult blood test, has shown to correlate with the mortality of colon cancer. The FOBT positive result might relate to the reduction of mortality to colon cancer [@mandel1993reducing]. I found several classes of bacteria that has a higher abundance in the FOBT positive samples, than the FOBT negative sample, which can be used as another additional screening method for colon cancer. For example, in both adenoma and carcinoma samples, Fusobacteriia and Deltaproteobacteria have found more abundant in FOBT positive sample. In contrast, Gammaproteobacteria has been found to have lower abundance in FOBT positive samples (Figure 2). This observation has very much related to the function of the bacteria, which then can even help us provide a more accurate treatment. *Fusobacterium*, the Genus within Class Fusobacteria has been found as a possible opportunistic pathogen located in both mouth and gut that is implicated in inflammatory diseases [@signat2011role]. During the cancer progression especially if the lesion has caused the bleeding, *Fusobacterium* might participate in inducing the inflammation in the colon. Thus, it is possible to use anti-inflammatory medication to help reduce the syndrome. 

There are also a lot of factors other than FOBT test that plays a role the colon cancer progression. I have investigated how the Body Mass Index (BMI), the gender, geographical location and, medication will impact the microbiome distribution pattern. For the BMI, there is no obvious correlation with the colon cancer development nor the change in the microbiome community. As for gender, Fusobacteria is only found in FOBT positive male samples with lower abundance in adenoma than carcinoma (Figure 3). In combination with the previous information, having less abundance in adenoma indicate less inflammation is happening in precancerous condition. Also, Fusobacteria will more likely to be the cause of inflammation in male than female. For Negativicutes, it is more commonly found in female adenoma than male, but it dramatically decreases in the female carcinoma samples, while only have a slight decrease in male carcinoma samples. Negativicutes is known to aid in the production or conversion of either succinate or lactate to propionate in the human gut [@reichardt2014phylogenetic]. The intake of propoinate in the colon, will reduce energy intake and prevent weight gain [@chambers2015effects]. Thus, females are more vulnerable to weight loss, when they develop colon carcinoma.

Family Lachnospiraceae has been known to help maintain the gut health as active plant degraders [@biddle2013untangling]. All the Genus in the Family has decreased abundance in the FOBT postive samples, but not necessary in adenoma and carcinoma (Figure 4). This is an indication that the FOBT positive is a relatively strong reflection of the decrease of the health of the colon, in colon cancer progression. Thus, if a patient has a decrease in the abundance of Lachonospiraceae, then extra attention is needed to maintain gut health and function. *Ruminococcus* is the Genus that belongs to the Family Lachnospiraceae, which has a particular geographic distribution that relate to the colon cancer progression. *Ruminococcus* is only commonly found in adenoma samples of patients that are from Toronto, Canada. *Ruminococcus* is only found in carcinoma samples from patients in Boston and Huston, USA. In contrast, *Ruminococcus* is found in normal samples for people from all four locations (Figure 5). This is an indication that different diet at different geographic locations will somehow shape the microbiome in adenoma and carcinoma cancer progressions. This also an indication that, while using microbiome as a screening tool to predict and treat colon cancer, we have to consider both progression stages and the region the patients come from. To provide more accurate prediction and more effective treatments. What's more, it also possible that different dietary restirction or locations can cause various mutations, which might induce the chance of getting colon cancer and alter the microbiome differently [@janout2001epidemiology]. 

Lastly, the medication will alternate the microbiome composition as well. Lipitor is a medication commonly used by patients in the experiment. Lipitor is the brand name for Atorvastatin, which has commonly known to be used for lowing blood cholesterol and prevent cardiovascular disease. However, it has recently been found to prevent colon carcinogenesis [@wu2015chemopreventive]. This matches up with my observation that *Ruminococcus* abundance under different treatment effects was only available for adenoma samples, Lipitor might be used to prevent the colon cancer progression to carcinoma (Figure 6). Also, I investigated how different medication combination with Lipitor will impact the *Ruminococcus* abundance. The *Ruminococcus* remains at a relatively high level across treatments and gender. Except for male individual treated norvasc, methoformim and allupurinal (Figure 6). The medications are not specific to colon cancer, There might be a side effect of combing different medications, which lead to the decrease of the *Ruminococcus*  abundance. However, this needs more investigation on the effects of medications on each other. 

There are several limitations of this paper - the major one being that all the prediction and investigation on the microbiome community are based on fecal samples. However, there is a possibility that in the colon cancer tissues there are some unique bacteria than the fecal samples. For carcinoma patients, we do not have the stage information, which can be very helpful in order to provide a more detailed analysis. There is also some more information we can collect from patients, like whether they have dietary restriction, and what are the common types of food they consume on a daily basis are. For the medication information, some medications are not targeting colon cancer. Therefore, information about some other disease the patients have might be helpful to make more sense of the mediation combination and their effects on microbiome composition. 

In summary, in order to use the distribution pattern and abundance change of microbiomes to predict and treat different progression stages of the colon cancer, there are many parameters need to be taken into consideration. Also, we need a sufficient amount of understanding of bacteria functions in the colon. As a result, using gender, geographical information can help us identify some key species of bacteria and further understand the relationship with the colon cancer. It is highly possible that we can develop a module in incorporate all the parameters to predict the development and risk of colon cancer or to provide more accurate treatment to improve the survival rate of the colon cancer. What's more, the same method can also be applies to other types of disease for alternative treatments.







# Sources Cited
